src/main/groovy/wooga/gradle/appcenter/api/AppCenterRest.groovy

```diff
@@ -0,0 +1,44 @@
+package wooga.gradle.appcenter.api
+
+import org.apache.http.HttpEntity
+import org.apache.http.HttpResponse
+import org.apache.http.client.HttpClient
+import org.apache.http.client.methods.HttpPost
+import org.apache.http.entity.mime.MultipartEntityBuilder
+import org.apache.http.entity.mime.content.FileBody
+import org.gradle.api.GradleException
+
+import java.util.logging.Logger
+
+class AppCenterRest {
+    static Logger logger = Logger.getLogger(AppCenterRest.name)
+
+    static Boolean uploadResources(HttpClient client, String apiToken, String uploadUrl, File binary, Integer retryCount = 0, Long retryTimeout = 0) {
+        HttpPost post = new HttpPost(uploadUrl)
+        FileBody ipa = new FileBody(binary)
+        post.setHeader("X-API-Token", apiToken)
+        HttpEntity content = MultipartEntityBuilder.create()
+                .addPart("ipa", ipa)
+                .build()
+
+        post.setEntity(content)
+        HttpResponse response = client.execute(post)
+
+        if (response.statusLine.statusCode >= 500) {
```

Comment:
I wonder if it should be in case of any error, also client errors (in case internet connection drops etc), not only 500+?

Reply:
At the moment we only see 500is errors causing our hicups. If you know a few good http error codes so I don't have to probe the result body I can add them. But retries can be problematic when their are implemented as dumb as I did here.